### PR TITLE
goto-gcc: run original compiler even when output is /dev/null

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

File: regression/goto-gcc/dev_null1/main.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+void gcc_fails(void)
+{
+  asm volatile ( ".if ((1 > 0) < 0); .error \"this is gcc\";.endif" );
```
Comment (smowton) [inline regression/goto-gcc/dev_null1/main.c]: What is this peculiar magic test?

Comment (tautschnig) [reply regression/goto-gcc/dev_null1/main.c]: It's modeled after code that Xen uses to figure out properties of the underlying assembler...

Comment (smowton) [reply regression/goto-gcc/dev_null1/main.c]: Would 
```
#if defined(__GNUC__) && !defined(__clang__)
#error "GCC fails"
#endif
```
achieve the same effect?

Comment (tautschnig) [reply regression/goto-gcc/dev_null1/main.c]: I don't think so, because it would (correctly) fail even without this patch.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 00a2cecbe6f292af4c73ba23e61f16c74286e21e).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78305481

File: regression/goto-gcc/dev_null1/main.c (hunk 1 above)
Comment (smowton) [reply regression/goto-gcc/dev_null1/main.c]: OK, why do you need a separate thing that makes GCC and Clang fail? Would a single ASM block that always `.error`s do?

I'm guessing the idea is we're looking for something that trips the native compiler but not goto-cc, which I guess ignores the odd ASM block?

Comment (tautschnig) [reply regression/goto-gcc/dev_null1/main.c]: True, a single block would do. Will simplify.

Comment (tautschnig) [reply regression/goto-gcc/dev_null1/main.c]: Done.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: ea9dc15f36ec553fe256b030f442e970e0abb9c8).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78477568

**Review by smowton — APPROVED** [review]

**Review by kroening — APPROVED** [review]

**Pull request merged**